### PR TITLE
emacs interface can now amend commits

### PR DESCRIPTION
**Pull request by dkogan**
Hi. I've been using a half-assed version of this for years. Here's a full-assed version. In its most basic form you can now C-c C-S-c to amend the last commit instead of creating a new one. You can also invoke this with a new function or by invoking the old function with a prefix arg.

### CODE REVIEW

File: commit-patch-buffer.el [hunk 1]
```diff
@@ -84,16 +90,30 @@ DIRECTORY with commit-patch(1)."
...
+
+(defun commit-patch-buffer-amend (&optional arg)
+  "Commit the patch in the current buffer, applying it to the
+repository in the appropriate directory with commit-patch(1). If
+the current buffer is not in diff-mode or ARG is non-nil then it
+will ask interactively which buffer to commit and to which
+directory to commit it.  This is identical to
+`commit-patch-buffer' except it amends the last commit by default
+instead of creating a new one."
+  (interactive "P")
```
Comment (radford) [inline commit-patch-buffer.el]: Is this "P" needed?

**Comment by radford** [comment]
Thanks Dima.  This looks looks fine, though if you end up amending, you might fix the typo in the commit message.

File: commit-patch-buffer.el (hunk 1 above)
Comment (dkogan) [inline commit-patch-buffer.el]: Yes. It allows a prefix argument to affect commit-patch-buffer-amend in the same way it affects commit-patch-buffer.

**Comment by dkogan** [comment]
Typo fixed in my master branch. I don't see a github button to update this request, so get the patch from my tree.

**Comment by radford** [comment]
It's fixed automatically when you force a push, see the 5bfe812 update right above your comment.

**Comment by caldwell** [comment]
Thanks, Dima. Just yesterday I was wishing I could do this in Emacs without having to pop out to the command line…
I made just a couple minor tweaks in 00d997ec and 4659f112.

**Comment by dkogan** [comment]
David Caldwell <notifications@github.com> writes:

> Thanks, Dima. Just yesterday I was wishing I could do this in Emacs
> without having to pop out to the command line… I made just a couple
> minor tweaks in 00d997ec and 4659f112.

Ah yes. 00d997ec is what I wanted to do, but not quite fluent enough in
elisp yet. Thanks for cleaning that up. If you could update the .deb
once the jessie freeze is over, that'd be great too; let me know if you
need help with that.

**Pull request merged**